tests/catch/unit/stream/hipStreamWaitEvent.cc

```diff
@@ -74,6 +93,35 @@ __global__ void waitKernel(int clockRate, int seconds) {
   }
 }
 
+TEST_CASE("Unit_hipStreamWaitEvent_Default") {
+  hipStream_t stream{nullptr};
+  hipEvent_t waitEvent{nullptr};
+
+  HIP_CHECK(hipStreamCreate(&stream));
+  HIP_CHECK(hipEventCreate(&waitEvent));
+
+  REQUIRE(stream != nullptr);
+  REQUIRE(waitEvent != nullptr);
+
+  hipDeviceProp_t prop{};
+  HIP_CHECK(hipGetDeviceProperties(&prop, 0));
```

Comment:
Instead of using hard-coded 0th device, get the default device using hipGetDevice & use it in hipGetDeviceProperties.

Reply:
@rorake Code is refactored according to the comments. Please review.